sdk/messaging/azservicebus/admin/admin_client_models.go

```diff
@@ -16,3 +22,87 @@ const (
 	// EntityStatusReceiveDisabled indicates that an entity cannot be used for receiving.
 	EntityStatusReceiveDisabled EntityStatus = "ReceiveDisabled"
 )
+
+type (
+	// AccessRight is an access right (Manage, Send, Listen) for an AuthorizationRule.
+	AccessRight string
+
+	// AuthorizationRule is a rule with keys and rights associated with an entity.
+	AuthorizationRule struct {
+		// AccessRights for this rule.
+		AccessRights []AccessRight
+
+		// KeyName for this rule.
+		KeyName *string
+
+		// CreatedTime for this rule.
+		CreatedTime *time.Time
+
+		// ModifiedTime for this rule.
+		ModifiedTime *time.Time
+
+		// PrimaryKey for this rule.
+		PrimaryKey *string
+
+		// SecondaryKey for this rule.
+		SecondaryKey *string
+	}
+)
+
+const (
+	// AccessRightManage allows changes to an entity.
+	AccessRightManage = "Manage"
+	// AccessRightSend allows you to send messages to this entity.
+	AccessRightSend = "Send"
+	// AccessRightListen allows you to receive messages from this entity.
+	AccessRightListen = "Listen"
+)
+
+func internalAccessRightsToPublic(internalRules []atom.AuthorizationRule) []AuthorizationRule {
```

Comment:
Are we using a generated ATOM layer now, which is why we need to perform these translations?